metricbeat-tests/runner_test.go

```diff
@@ -133,6 +143,8 @@ func (mts *MetricbeatTestSuite) installedUsingConfiguration(version string, conf
 		return err
 	}
 
+	mts.configurationFile = configurationFilePath
```

Comment:
If we are storing the configuration file in an attribute of `mts` then we wouldn't need to pass it to `mts.runMetricbeatService()`.

Reply:
you are totally right!

Reply:
Mmm, thinking out loud, no. We just want to remove it when it's downloaded. Otherwise it will try to remove those in the workspace

Reply:
It wouldn't change when the file is removed, right? I only mean to modify `runMetricbeatService` so it takes the configuration file from the `configurationFile` attribute.

Reply:
But if we set the mts.configurationFile at some point, then the CleanUp method will remove it

Reply:
Yes, but this also happens now...

Reply:
It only removes the file for the `installedUsingConfiguration()` method, which is the only one setting that attribute.

If we use the attribute inside `runMetricbeatService()` then it must be set also here: `installedAndConfiguredForModule` (the method using default files). Therefore, it would be removed on cleanUp even for the second option

Reply:
Ok, I get your point, you are right, sorry.

Maybe we should make explicit that setting `mts.configurationFile` is only intended for cleanup.

Reply:
Or maybe we should create a temporary dir that gets removed on cleanup and place there all temporary files a test may need.

Reply:
I like the idea!

Reply:
Maybe in another PR?

Reply:
Sure, you already had my approval :slightly_smiling_face:

Reply:
I added two commits: one making explicit the cleanUp of tmp files, and another one fixing a regression based by my previous pull